rts/game_MC/lua/unit_factory.lua

```diff
@@ -371,28 +408,29 @@ function rts_unit_factory.init_workshop()
     --[[attr]]UnitAttr.ATTR_NORMAL,
     --[[attack_multiplier]]attack_multiplier,
     --[[cant_move_over]]cant_move_over,
-    --[[build_from]]UnitType.WORKER,
+    --[[build_from]]UnitType.PEASANT,
     --[[build_skills]]build_skills
   )
   return ut
 end
 
 
-function rts_unit_factory.init_defense_tower()
+function rts_unit_factory.init_guard_tower()
   local allowed_cmds = {CmdType.ATTACK}
   local attack_multiplier = {
-    {UnitType.WORKER, 1.0},
-    {UnitType.SOLDIER, 1.0},
-    {UnitType.TRUCK, 1.0},
-    {UnitType.TANK, 1.0},
-    {UnitType.CANNON, 1.0},
-    {UnitType.FLIGHT, 1.0},
+    {UnitType.PEASANT, 1.0},
+    {UnitType.SWORDMAN, 1.0},
+    {UnitType.SPEARMAN, 1.0},
+    {UnitType.CAVALRY, 1.0},
+    {UnitType.ARCHER, 1.0},
+    {UnitType.DRAGON, 1.0},
+    {UnitType.CATAPULT, 1.0},
     {UnitType.BARRACK, 1.0},
-    {UnitType.FACTORY, 1.0},
-    {UnitType.HANGAR, 1.0},
+    {UnitType.BLACKSMITH, 1.0},
+    {UnitType.STABLES, 1.0},
     {UnitType.WORKSHOP, 1.0},
-    {UnitType.DEFENSE_TOWER, 1.0},
-    {UnitType.BASE, 1.0}
+    {UnitType.GUARD_TOWER, 1.0},
+    {UnitType.TOWN_HALL, 1.0}
   }
```

Comment:
note we designed defense tower should be outranged by catapult

Reply:
yeah, we should set appropriate att_r for catapult. set it 7 for catapult and 5 for guard tower.